Sources/SpecialErrorPages/SSLErrorType.swift

```diff
@@ -17,34 +17,32 @@
 //
 
 import Foundation
+import WebKit
 
-public enum SSLErrorType: String {
+public let SSLErrorCodeKey = "_kCFStreamErrorCodeKey"
+
+public enum SSLErrorType: String, Encodable {
 
     case expired
-    case wrongHost
     case selfSigned
+    case wrongHost
     case invalid
 
-    public static func forErrorCode(_ errorCode: Int) -> Self {
-        switch Int32(errorCode) {
-        case errSSLCertExpired:
-            return .expired
-        case errSSLHostNameMismatch:
-            return .wrongHost
-        case errSSLXCertChainInvalid:
-            return .selfSigned
-        default:
-            return .invalid
+    init(errorCode: Int) {
+        self = switch Int32(errorCode) {
+        case errSSLCertExpired: .expired
+        case errSSLXCertChainInvalid: .selfSigned
+        case errSSLHostNameMismatch: .wrongHost
+        default: .invalid
         }
     }
 
-    public var rawParameter: String {
-        switch self {
-        case .expired: return "expired"
-        case .wrongHost: return "wrong_host"
-        case .selfSigned: return "self_signed"
-        case .invalid: return "generic"
-        }
-    }
+}
 
+extension WKError {
+    public var sslErrorType: SSLErrorType? {
```

Comment:
Out of curiosity, as WKNavigationDelegate gives us an NSError should we make this an extension on NSError rather than on WKError? In that way there’s no need to create the WKError with the NSError.